core/src/test/java/uk/gov/gchq/koryphe/impl/function/IterableFlattenTest.java

```diff
@@ -59,6 +60,19 @@ public void shouldFlattenIterableNumbers() {
         assertThat(result).isEqualTo(15);
     }
 
+    @Test
+    public void shouldFlattenIterableNumbersNull() {
+        // Given
+        final IterableFlatten<Number> function = new IterableFlatten<>(new Sum());
+        final List<Number> input = Lists.newArrayList((Number) null, (Number) null, (Number) null, (Number) null);
```

Comment:
Could be called 'shouldFlattenIterableNumbersOnlyNull' or 'shouldFlattenIterableNumbersAllNull'? There is no need for the superfluous casts of `(Number) null`, this is done implicitly.